src/Worker.Extensions.DurableTask/Exceptions/DurableSerializationException.cs

```diff
@@ -11,14 +11,20 @@ internal class DurableSerializationException : Exception
     // We set the base class properties of this exception to the same as the parent, 
     // so that methods in the worker after this can still (typically) access the same information vs w/o
     // this exception type. 
-    internal DurableSerializationException(Exception fromException) : base(fromException.Message, fromException.InnerException)
+    internal DurableSerializationException(Exception fromException) : base(CreateExceptionMessage(fromException), fromException.InnerException)
```

Comment:
Just want to understand why this change was made - the reason that I only did the JSON serialization in ToString() originally was because I didn't want to alter the actual properties of the exception - before this change, if the functions worker needed to access the properties of the user code exception, they were functionally unmodified, and it wasn't until the exception was converted to a string for gRPC transmission that the serialization happened. 
Can you help me understand what was the negative effect of this decision and what does serializing the message property improve?

Reply:
Understood. 
I think what I was missing was this:
> For some reason, the ToString() is only called by grpc log, while the grpc worker only pass exception message and inner Failure
Very surprising, TBH. 

When I was talking about the "properties of the user code exception" I meant the properties of the exception exposed to the ooproc worker - before my change that added DurableSerializationException, the native user code exception type was thrown all the way up until the gRPC logging step. I was trying to be extra careful not to change behavior of the worker, especially for "special case" exception types like OutOfMemoryException. 

Anyway, based on the test case (which looks great, btw) and your own experimentation, I'm good to approve this PR. Thanks for investigating this!